pipeline/src/client.py

```diff
@@ -259,6 +263,27 @@ def reprocess(self, start_date=None, end_date=None):
     def create_all_views(self):
         return self.flagged.create_views_all_flags()
 
+    ###########################################################
+
+    def init_flag_dict(self):
+        self._flag_lookup = dict()
+        df = self.flags.get_full_table()
```

Comment:
In regards to error handling on this method, one issue is that table.py handles all errors, so all I will know is whether None or a DataFrame was returned. That makes it more difficult for the client to sanely judge what action should be taken. I was of the mind to treat it as an "empty or non-existant table", as that has the fewest side-effects that I could think of. I can added logger/print output to this as well. Thoughts on this?

If the structure of the local code changes and the database does not get updated, this method should still work if it does not get changed as well I believe. Since the column attributes are named exclusively, it should remain consistent with the table (as columns should never get renamed/removed). Thus, for the purpose of servicing name-to-id queries, I felt this was fairly sturdy. Please set me straight if I didn't understand your original comment correctly.

Reply:
I completely forgot about the first point. In that case, since we can't really enforce an initialized flag_lookup as a post condition of this method, I would argue that the --flag code should have some logic to ensure that this field is set before trying to utilize it.

My original comment came about from me getting that error due to having the old table without the name field, and also when the table does not exist. At this point, I think that the method could stand to try and create the table if it can't get the table, so at least the pipeline can then auto-update the flags table as more are added. It wouldn't address the more irregular issues like a bad engine but it would improve the ease of use and ensure that flag_lookup is always up to date.

Reply:
Gotcha, I'll add the attempt to create the table tomorrow. Currently, (line 20 ArgInterface), the absence of the table is not differentiated from the absence of any particular flag. I'll add a different log message for those two conditions.

Reply:
Oh I forgot about client_instance.lookup_flag_id() too - can you tell the burn out from covid remote/quarantine is real :joy: